bdk_chain/src/tx_data_traits.rs

```diff
@@ -7,14 +7,14 @@ use bitcoin::{Block, OutPoint, Transaction, TxOut};
 ///
 /// We would prefer just work with things that can give us a `Iterator<Item=(OutPoint, &TxOut)>`
 /// here but rust's type system makes it extremely hard to do this (without trait objects).
-pub trait ForEachTxout {
-    fn for_each_txout(&self, f: &mut impl FnMut((OutPoint, &TxOut)));
+pub trait ForEachTxOut {
+    fn for_each_txout(&self, f: impl FnMut((OutPoint, &TxOut)));
 }
 
-impl ForEachTxout for Block {
-    fn for_each_txout(&self, f: &mut impl FnMut((OutPoint, &TxOut))) {
+impl ForEachTxOut for Block {
+    fn for_each_txout(&self, mut f: impl FnMut((OutPoint, &TxOut))) {
         for tx in self.txdata.iter() {
-            tx.for_each_txout(f)
+            tx.for_each_txout(&mut f)
```

Comment:
TIL `FnMut` is implemented for `&mut F where F: FnMut`.

Reply:
I changed the trait to take in a closure without a reference... :/